dora-parser/src/parser.rs

```diff
@@ -235,11 +245,20 @@ impl<'a> Parser<'a> {
         let mut methods = Vec::new();
 
         while !self.token.is(TokenKind::RBrace) {
-            let modifiers = self.parse_annotations()?;
-            let mods = &[Modifier::Static, Modifier::Internal, Modifier::Cannon];
-            self.restrict_modifiers(&modifiers, mods)?;
-
-            methods.push(self.parse_function(&modifiers)?);
+            let modifiers = self.parse_annotation_usages()?;
```

Comment:
Why is this logic here reverted? `restrict_modifiers` only allows `@static`, `@internal` or `@cannon`. Each time we add an annotation this would have to be updated.

Reply:
I'm trying to avoid a situation where we have to go back to making a distinction between "our special (modifier) annotations" and "annotations defined in libraries/by users".

You are right that using a blacklist here looks bad, because when looking at the set of "special" modifiers it's a lot more. But when looking at the set of "all annotations", we cannot use a whitelist, as we cannot whitelist annotations we don't even know (because they might be defined in a third-party library somewhere in the future).

My plan is to address this down the road by pulling this logic out of the compiler altogether and move it to the annotation definitions such that annotations can define where they can be placed themselves, e. g.:

```
@target(CLASS, CLASS_METHOD)
annotation abstract

@target(CLASS_METHOD)
annotation final

@target(CLASS, CLASS_METHOD, CLASS_VALUE, CLASS_VARIABLE, TRAIT, TRAIT_METHOD, IMPL, ...)
annotation deprecated(...)

@target(CLASS, CLASS_METHOD)
annotation deprecatedOpen(...)

@target(ANNOTATION)
annotation target(...)
```

Is this ok for you?

Reply:
What would keeping this simple entail?

Reply:
Not having to manually add annotations at all possible usages.

Thinking about this and if we already touch the logic, I think it might be best to just allow any annotation here in the parser and check annotations later during semck.

Reply:
> Not having to manually add annotations at all possible usages.

Yes, that was my intent for moving the determination where annotations can be placed to the individual annotation's declaration itself.

> Thinking about this and if we already touch the logic, I think it might be best to just allow any annotation here in the parser and check annotations later during semck.

Agreed, that was the plan after I get `@target` working (which involves properly dealing with annotation arguments first).
Semck would then simply check the allowed places (from `@target`) against the actual location.